src/ucs/debug/log.c

```diff
@@ -120,90 +120,85 @@ size_t ucs_log_get_buffer_size()
+        start_idx = ucs_log_file_last_idx - 1;
+    } else {
+        start_idx = ucs_log_file_last_idx;
+        ucs_log_file_last_idx++;
+    }
 
-        if (ucs_global_opts.log_file_backup_count != 0) {
-            if (ucs_log_file_last_idx == ucs_global_opts.log_file_backup_count) {
-                /* remove the last file and log rotation from the
-                 * `log_file_backup_count - 1` file */
-                ucs_log_get_file_name(old_log_file_name,
-                                      UCS_LOG_MAX_FILE_NAME,
-                                      ucs_log_file_last_idx);
-                unlink(old_log_file_name);
-                start_idx = ucs_log_file_last_idx - 1;
-            } else {
-                start_idx = ucs_log_file_last_idx;
-                ucs_log_file_last_idx++;
-            }
+    ucs_assert(ucs_log_file_last_idx <= ucs_global_opts.log_file_rotate);
+    ucs_assert(start_idx <= (ucs_global_opts.log_file_rotate - 1));
 
-            ucs_assert(ucs_log_file_last_idx <=
-                       ucs_global_opts.log_file_backup_count);
-            ucs_assert(start_idx <= (ucs_global_opts.log_file_backup_count - 1));
+    for (idx = start_idx; idx != UINT_MAX; --idx) {
+        ucs_assert(idx <= start_idx);
 
-            for (idx = start_idx; idx != UINT_MAX; --idx) {
-                ucs_assert(idx <= start_idx);
+        ucs_log_get_file_name(old_log_file_name,
+                              UCS_LOG_MAX_FILE_NAME, idx);
+        ucs_log_get_file_name(new_log_file_name,
+                              UCS_LOG_MAX_FILE_NAME, idx + 1);
 
-                ucs_log_get_file_name(old_log_file_name,
-                                      UCS_LOG_MAX_FILE_NAME, idx);
-                ucs_snprintf_zero(new_log_file_name, UCS_LOG_MAX_FILE_NAME,
-                                  "%s.%u", ucs_log_file_names[idx], idx + 1);
+        ucs_assertv_always(access(old_log_file_name, W_OK) == 0,
+                           "unable to write to %s", old_log_file_name);
 
-                ucs_assertv_always(access(old_log_file_name, W_OK) == 0,
-                                   "unable to write to %s", old_log_file_name);
+        /* coverity[toctou] */
+        ret = rename(old_log_file_name, new_log_file_name);
+        if (ret) {
+            ucs_fatal("failed to rename %s to %s: %m",
+                      old_log_file_name, new_log_file_name);
+        }
 
-                /* coverity[toctou] */
-                ret = rename(old_log_file_name, new_log_file_name);
-                if (ret) {
-                    ucs_fatal("failed to rename %s to %s: %m",
-                              old_log_file_name, new_log_file_name);
-                }
+        ucs_assertv_always(access(old_log_file_name, F_OK) == -1,
+                           "%s must not exist", old_log_file_name);
+        ucs_assertv_always(access(new_log_file_name, W_OK) == 0,
+                           "unable to write to %s", new_log_file_name);
```

Comment:
maybe ucs_fatal instead of assert (it's not programming error, but a system issue)

Reply:
done